### PR TITLE
Fix type error while saving edits to System Prompt

### LINKED ISSUE #291: Type error while saving edits to System Prompt
Error seems to be pointing to this line 

https://github.com/tarasglek/chatcraft.org/blob/0b0ee516006a825ae18606b2d2390e4b55e5ddb3/src/components/Message/MessageBase.tsx#L207

`TypeError: Cannot read properties of null (reading 'removeAttribute')` is thrown like so

Screenshot 2023-11-15 at 9 04 56 AM

### PR DESCRIPTION
**Pull request by DukeManh**
Fix #291 and add keyboard event handler hook to reuse keyboard event check logic

### CODE REVIEW

**Comment by DukeManh** [comment]
I noticed 'Cmd + Enter' on message edit form wasn't working on MAC due to my previous PR. I fixed that and added a hook to extract the keyboard event check logic.

File: src/components/PromptForm/MobilePromptForm.tsx [hunk 1]
```diff
@@ -3,12 +3,12 @@ import { Box, chakra, Flex } from "@chakra-ui/react";
 import AutoResizingTextarea from "../AutoResizingTextarea";
 
 import { useSettings } from "../../hooks/use-settings";
-import { isMac, isWindows } from "../../lib/utils";
 import NewButton from "../NewButton";
 import MicIcon from "./MicIcon";
 import { isTranscriptionSupported } from "../../lib/speech-recognition";
 import PromptSendButton from "./PromptSendButton";
 import AudioStatus from "./AudioStatus";
+import { useKeyDownHandler } from "../../hooks/use-key-down-handler";
```
Comment (humphd) [inline src/components/PromptForm/MobilePromptForm.tsx:11]: Do we need this in mobile?  I guess people could use a keyboard with their small screen.

File: src/hooks/use-key-down-handler.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useCallback, KeyboardEvent } from "react";
```
Comment (humphd) [inline src/hooks/use-key-down-handler.ts:1]: This is great, thanks for pulling this out.

**Review by humphd — APPROVED** [review]
Looks great, land it when you're ready

File: src/components/PromptForm/MobilePromptForm.tsx (hunk 1 above)
Comment (DukeManh) [reply src/components/PromptForm/MobilePromptForm.tsx:11]: Yes I think it's for mobiles and small or resized screens.

**Pull request merged**